src/jit/block.h

```diff
@@ -1110,6 +1110,53 @@ struct flowList
     {}
 };
 
+// This enum represents a pre/post-visit action state to emulate a depth-first
+// spanning tree traversal of a tree or graph.
+enum DfsStackState
+{
+    DSS_Invalid,      // The initialized, invalidate error state
+    DSS_Pre,          // The DFS pre-order (first visit) traversal state
+    DSS_Post          // The DFS post-order (last visit) traversal state
+};
+
+// This struct represents an entry in a stack used to emulate a non-recursive
+// depth-first spanning tree traversal of a graph. The entry contains a block
+// pointer which is more useful for some algorithms.
+struct DfsBlockEntry
+{
+    DfsStackState       dfsStackState;  // The pre/post traversal action for this entry
+    BasicBlock*         dfsBlock;       // The corresponding block for the action
+
+    DfsBlockEntry()
+        : dfsStackState(DSS_Invalid)
+        , dfsBlock(nullptr)
+    {}
+
+    DfsBlockEntry(DfsStackState state, BasicBlock* basicBlock)
+        : dfsStackState(state)
+        , dfsBlock(basicBlock)
+    {}
+};
+
+// This struct represents an entry in a stack used to emulate a non-recursive
+// depth-first spanning tree traversal of a graph. The entry contains a block
+// number which is more useful for some algorithms.
+struct DfsNumEntry
```

Comment:
This comment and the one above differ by a single word (pointer/number). Maybe you could just have a single comment for both of them, which points out that difference.